.github/workflows/sdks.yml

```diff
@@ -50,17 +52,38 @@ jobs:
             SWIFT_BRANCH="development"
           fi
           cd
-          SWIFT_TAG="${{ steps.check.outputs.latest-tag }}"
           wget -q https://download.swift.org/$SWIFT_BRANCH/ubuntu2204/$SWIFT_TAG/$SWIFT_TAG-ubuntu22.04.tar.gz
           echo "got latest toolchain: ${SWIFT_TAG}"
+      - name: Get cached macOS toolchain
+        id: cache-toolchain-macos
+        uses: actions/cache@v4
+        with:
+          path: ~/${{ steps.check.outputs.latest-tag }}-osx.pkg
+          key: swift-macos-${{ steps.check.outputs.latest-tag }}-toolchain
+      - name: Download macOS toolchain
+        if: ${{ steps.cache-toolchain-macos.outputs.cache-hit != 'true' }}
+        run: |
+          SWIFT_TAG="${{ steps.check.outputs.latest-tag }}"
+          if [[ ${{ matrix.version }} = 'release'* ]]; then
+            # downcase "swift-5.10.1-RELEASE" to "swift-5.10.1-release"
+            SWIFT_BRANCH=$(echo "$SWIFT_TAG" | tr '[:upper:]' '[:lower:]')
+          elif [ ${{ matrix.version }} = 'devel' ]; then
+            SWIFT_BRANCH="swift-6.0-branch"
+          else
+            SWIFT_BRANCH="development"
+          fi
+          cd
+          wget -q https://download.swift.org/$SWIFT_BRANCH/xcode/$SWIFT_TAG/$SWIFT_TAG-osx.pkg
```

Comment:
Rather than duplicate all this logic here, simply move this one command that downloads this mac toolchain to right after the `wget` command that downloads the Ubuntu toolchain above, putting each inside a `steps.cache-toolchain-macos.outputs.cache-hit` or linux check, and only running that download step if either one of those cache hits is not true, ie `if: ubuntu.cache-hit != true || macos.cache-hit != true`.